dango/vesting/src/query.rs

```diff
@@ -0,0 +1,70 @@
+use {
+    crate::POSITIONS,
+    dango_types::vesting::{ClaimablePosition, PositionIndex, QueryMsg},
+    grug::{Addr, Bound, ImmutableCtx, Json, JsonSerExt, Order, StdResult},
+    std::collections::BTreeMap,
+};
+
+const DEFAULT_PAGE_LIMIT: u32 = 30;
+
+#[cfg_attr(not(feature = "library"), grug::export)]
+pub fn query(ctx: ImmutableCtx, msg: QueryMsg) -> StdResult<Json> {
+    match msg {
+        QueryMsg::Position { idx } => {
+            let res = position(ctx, idx)?;
+            res.to_json_value()
+        },
+        QueryMsg::Positions { start_after, limit } => {
+            let res = positions(ctx, start_after, limit)?;
+            res.to_json_value()
+        },
+        QueryMsg::PositionsByUser {
+            user,
+            start_after,
+            limit,
+        } => {
+            let res = positions_by_user(ctx, user, start_after, limit)?;
+            res.to_json_value()
+        },
+    }
+}
+
+fn position(ctx: ImmutableCtx, idx: PositionIndex) -> StdResult<ClaimablePosition> {
+    POSITIONS
+        .load(ctx.storage, idx)
+        .map(|val| val.with_claimable_amount(ctx.block.timestamp))
+}
+
+fn positions(
+    ctx: ImmutableCtx,
+    start_after: Option<PositionIndex>,
+    limit: Option<u32>,
+) -> StdResult<BTreeMap<PositionIndex, ClaimablePosition>> {
+    let start = start_after.map(Bound::Exclusive);
+    let limit = limit.unwrap_or(DEFAULT_PAGE_LIMIT) as usize;
+
+    POSITIONS
+        .range(ctx.storage, start, None, Order::Ascending)
+        .take(limit)
+        .map(|res| res.map(|(k, v)| (k, v.with_claimable_amount(ctx.block.timestamp))))
+        .collect()
+}
+
+fn positions_by_user(
+    ctx: ImmutableCtx,
+    user: Addr,
+    start_after: Option<PositionIndex>,
+    limit: Option<u32>,
+) -> StdResult<BTreeMap<PositionIndex, ClaimablePosition>> {
+    let start = start_after.map(Bound::Exclusive);
+    let limit = limit.unwrap_or(DEFAULT_PAGE_LIMIT) as usize;
+
+    POSITIONS
+        .idx
+        .user
+        .prefix(user)
+        .range(ctx.storage, start, None, Order::Ascending)
+        .take(limit)
+        .map(|res| res.map(|(k, v)| (k, v.with_claimable_amount(ctx.block.timestamp))))
+        .collect()
+}
```

Comment:
these functions should be named `query_position`, `query_positions`, `query_positions_by_user` to be consistent with other contracts in the codebase